### PR TITLE
Add Settings Page

### PR DESCRIPTION
**Pull request by NickEckert**
- Added settings page
- Removed upload progress for now, will add in separate branch
- General code clean up

### CODE REVIEW

File: pilot/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 1]
```diff
@@ -47,6 +47,11 @@
       "idiom" : "iphone",
       "filename" : "Icon-App-60x60@3x.png",
       "scale" : "3x"
+    },
+    {
+      "idiom" : "ios-marketing",
+      "size" : "1024x1024",
+      "scale" : "1x"
```
Comment (RohanNagar) [inline pilot/Assets.xcassets/AppIcon.appiconset/Contents.json:54]: What is this icon?

File: pilot/LightningRouter.swift [hunk 2]
```diff
@@ -14,9 +14,9 @@ import HTTPStatusCodes
 /// The router builds static content related to the URL such as parameters, headers, etc..
 enum LightningRouter: URLRequestConvertible {
     
-    case publish(Platform, [String: String]?)
-    case extendToken(Platform)
-    case getOauthURL(Platform)
+    case publish(Platform, [String: Any]?)
```
Comment (RohanNagar) [inline pilot/LightningRouter.swift]: Do you want to make this `PlatformType` too?

File: pilot/LoginViewController.swift [hunk 3]
```diff
@@ -27,21 +27,25 @@ class LoginViewController: UIViewController {
...
         self.passwordField.text = "password"
+        
+        // Set a request adapter for future network calls
+        NetworkManager.sharedInstance.adapter = AuthAdapter()
     }
     
     func styleUI() {
         
-        // Button Style Properties
+        //vartton Style Properties
```
Comment (RohanNagar) [inline pilot/LoginViewController.swift]: Typo?

File: pilot/LoginViewController.swift [hunk 4]
```diff
@@ -72,17 +76,18 @@ class LoginViewController: UIViewController {
             // Make the request
             PilotUser.fetch(with: ThunderRouter.login(email, hashedPassword), onSuccess: { pilotUser in
                 
-                // Set a request adapter for future network calls
-                let token = AuthToken(email: email, password: hashedPassword)
-                NetworkManager.sharedInstance.adapter = AuthAdapter(authToken: token)
+                PilotConfiguration.PilotCredentials.email = pilotUser.email!
+                PilotConfiguration.PilotCredentials.password = pilotUser.password!
```
Comment (RohanNagar) [inline pilot/LoginViewController.swift:80]: The password is immediately hashed after the user logs in, right? Just want to check on that.

File: pilot/PilotConfiguration.swift [hunk 5]
```diff
@@ -22,6 +22,21 @@ struct PilotConfiguration {
...
+        
+        // Facebook
+        static let facebookRedirectURL = "fb896501930437172://authorize"
+        static let facebookRedirectScheme = "fb896501930437172"
+        static let facebookTokenParamKey = "access_token"
+        
+        // Twitter
+        static let twitterRedirectURL = ""
+        static let twitterRedirectScheme = ""
+        static let twitterTokenParamKey = ""
```
Comment (RohanNagar) [inline pilot/PilotConfiguration.swift:34]: These will probably change with the new OAuth path, just be aware of that.

File: pilot/PlatformManager.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+//
+//  Created by Nick Eckert on 9/23/17.
+//  Copyright © 2017 sanction. All rights reserved.
+//
+
+import Foundation
+
+// Keep track of all platforms and their state
+
+class PlatformManager {
```
Comment (RohanNagar) [inline pilot/PlatformManager.swift:13]: Nice, I like this class!

File: pilot/Uploadable.swift [hunk 7]
```diff
@@ -13,6 +13,14 @@ import Alamofire
 import ObjectMapper
 import SwiftyJSON
 
+import Foundation
+
+import Foundation
+import Alamofire
+import ObjectMapper
+import AlamofireObjectMapper
+import SwiftyJSON
```
Comment (RohanNagar) [inline pilot/Uploadable.swift]: A lot of these are duplicate imports

**Review by RohanNagar — CHANGES_REQUESTED** [review]
Love the new settings page. Just some small code things and a few questions.

**Comment by RohanNagar** [comment]
Can you also merge master into this branch to get the latest changes?

File: pilot/LightningRouter.swift (hunk 2 above)
Comment (NickEckert) [reply pilot/LightningRouter.swift]: Done

File: pilot/Assets.xcassets/AppIcon.appiconset/Contents.json (hunk 1 above)
Comment (NickEckert) [reply pilot/Assets.xcassets/AppIcon.appiconset/Contents.json:54]: Not sure. I've delete unused icons, maybe it was one of those.

File: pilot/LoginViewController.swift (hunk 3 above)
Comment (NickEckert) [reply pilot/LoginViewController.swift]: Fixed

File: pilot/LoginViewController.swift (hunk 4 above)
Comment (NickEckert) [reply pilot/LoginViewController.swift:80]: Yes, those values are actually the ones returned by Thunder!

File: pilot/PilotConfiguration.swift (hunk 5 above)
Comment (NickEckert) [reply pilot/PilotConfiguration.swift:34]: Noted, they should be easily changeable!

File: pilot/Uploadable.swift (hunk 7 above)
Comment (NickEckert) [reply pilot/Uploadable.swift]: Thanks, I thought I got all those

**Review by RohanNagar — APPROVED** [review]
LGTM, thanks!

**Pull request merged**